src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js

```diff
@@ -64,18 +68,29 @@ class AvailableOrganisationUnitsTree extends PureComponent {
         return orgUnits
     }
 
-    handleOrgUnitClick(event, orgUnit) {
-        if (!this.state.selected.includes(orgUnit.path)) {
-            this.setState({ selected: [orgUnit.path] })
-            if (this.props.onChange) {
-                const selectedOrganisationUnitSplitted = orgUnit.path.split('/')
-                const selectedOrganisationUnitId =
-                    selectedOrganisationUnitSplitted[
-                        selectedOrganisationUnitSplitted.length - 1
-                    ]
-                this.props.onChange(selectedOrganisationUnitId)
-            }
+    getOrgUnitIdFromPath(path) {
+        return path.split('/').pop()
+    }
+
+    handleOrgUnitClickSingle(event, orgUnit) {
+        if (this.state.selected.includes(orgUnit.path)) {
+            return
         }
+        const id = this.getOrgUnitIdFromPath(orgUnit.path)
+        this.setState({ selected: [orgUnit.path] })
+        this.props.onChange && this.props.onChange(id)
+    }
+
+    handleOrgUnitClickMulti(event, orgUnit) {
+        const { selected } = this.state
+        const { path } = orgUnit
+        const paths = selected.includes(path)
+            ? selected.filter(selectedPath => selectedPath !== path)
+            : [...this.state.selected, orgUnit.path]
```

Comment:
I assume you can just use path here instead of orgUnit.path

Reply:
Yep, agreed @ismay